src/core/pipe/pipe.filter.js

```diff
@@ -0,0 +1,21 @@
+import * as guard from '../infrastructure/guard';
+
+export default function (data, ctx, next) {
+	const model = ctx.model;
+	guard.notUndefined(model.filter, 'model.filter');
```

Comment:
I think this guard is needed because for core model `filter` property is not required is it?

User can use `pipe.filter` function only if he defined `model.filter`.

Unlike that `guard.function` placed below is needed to be check inside the model, i agree :)